### PR TITLE
Deal with all cache invalidation in EContactRepo

### PR DESCRIPTION
**Pull request by leogrim**
@stkem @raymondng 
This is mostly about caching, a couple of APIs changes but expect them to change again once I tie EmailAccounts and EContacts.

### CODE REVIEW

File: kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala [hunk 1]
```diff
@@ -136,52 +116,35 @@ class EContactRepoImpl @Inject() (
...
+          insertedContacts.foreach(invalidateCache)
+          insertedContacts
         }
+        case _ => toBeInserted.map(save)
       }
+    } catch { case ex: Throwable =>
+      log.error(s"Failed to insert mutiple contacts: $ex")
+      toBeInserted.map(save)
     }
-    log.info(s"[insertOnDupUpdate(${userId.id}, ${c.email})]")
```
Comment (leogrim) [inline kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala:166]: I'm going to try this in production to see how MySQL handles fancy Slick :)

File: kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala [hunk 2]
```diff
@@ -43,8 +37,8 @@ trait EContactRepo extends Repo[EContact] {
 class EContactRepoImpl @Inject() (
   val db: DataBaseComponent,
   val clock: Clock,
-  val econtactTypeaheadCache: EContactTypeaheadCache,
-  val econtactCache: EContactCache,
+  econtactCache: EContactCache,
+  mode: Mode,
```
Comment (raymondng) [inline kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala:41]: Nice :+1: (didn't know about this way back when)

Comment (leogrim) [inline kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala:41]: No worry, I think most people still don't know about this ;)

File: kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala (hunk 1 above)
Comment (raymondng) [inline kifi-backend/modules/abook/app/com/keepit/abook/EContactRepo.scala:166]: Interesting. You can also try this locally (or b01 if the instance still exist)

**Comment by raymondng** [comment]
Overall lgtm. Thanks!

**Pull request merged**